api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -491,3 +510,297 @@ model HcpOpenShiftClusterAdminCredential {
  *   End HCP cluster credentials
  * =======================================
  */
+
+/*
+ * =======================================
+ *  ExternalAuth resources
+ * =======================================
+ */
+
+/** External Auth profile */
+model ExternalAuthProperties {
+  /** Provisioning state */
+  @visibility(Lifecycle.Read)
+  provisioningState?: ExternalAuthProvisioningState;
+
+  /** An observation of the current state with additional information. */
+  @visibility(Lifecycle.Read)
+  condition?: ExternalAuthCondition;
+
+  /** Token Issuer profile */
+  issuer: TokenIssuerProfile;
+
+  /** External Auth OIDC clients
+   * There must not be more than 20 entries and entries must have unique namespace/name pairs.
+   */
+  @maxItems(20)
+  @OpenAPI.extension(
+    "x-ms-identifiers",
+    #["component", "clientId", "extraScopes"]
+  )
+  clients?: ExternalAuthClientProfile[];
+
+  /** External Auth claim
+   * This configures how claims are validated and applied.
+   */
+  claim: ExternalAuthClaimProfile;
+}
+
+/** Condition defines an observation of the external auth state. */
+model ExternalAuthCondition {
+  /** This is a PascalCase (or in foo.example.com/PascalCase) code to represent the type of condition. */
+  type: ExternalAuthConditionType;
+
+  /** The status of the condition. */
+  status: StatusType;
+
+  /** The last time the condition transitioned from one status to another. */
+  lastTransitionTime: utcDateTime;
+
+  /** This contains a programmatic identifier indicating the reason for the condition's last transition. */
+  @minLength(1)
+  @maxLength(1024)
+  reason: string;
+
+  /** This is a human readable message indicating details about the transition.
+   * This may be an empty string.
+   */
+  @minLength(1)
+  @maxLength(32768)
+  message: string;
+}
+
+/** Representation of the possible values of a external auths condition status. */
+union StatusType {
+  string,
+
+  /** Indicates that the condition status is True. */
+  True: "True",
+
+  /** Indicates that the condition status is False. */
+  False: "False",
+
+  /** Indicates that the condition status is unknown. */
+  Unknown: "Unknown",
+}
```

Comment:
Optional, but this is a case where's I think it's fine to define the enum inline.
```
  /** The status of the condition. */
  status: "True" | "False" | "Unknown" | string;
```
Everyone knows what these values mean.